### PR TITLE
Add resource names

### PR DESCRIPTION
**Pull request by michaelbausor**

### CODE REVIEW

**Comment by michaelbausor** [comment]
PTAL

**Comment by garrettjonesgoogle** [comment]
I think we need to avoid having multiple classes with the same name in the same repo, just under different packages.

**Comment by michaelbausor** [comment]
I agree, it is not very nice. We could:
- Rename our ResourceName interface (I'm struggling to come up with anything I like - ResourceFormat?)
- Move the existing ResourceName class back into gax
- Other?

**Comment by garrettjonesgoogle** [comment]
* Maybe `TypedResourceName`? It's not ideal though, because then you get `TypedResourceNameType` (stutter), and then `UntypedResourceName` implements `TypedResourceName`, which is weird. 
* Moving `ResourceName` into gax is just squishing things around, and I don't think makes the situation better. 
* Is there enough synergy between the two that would allow some kind of merging?

**Comment by michaelbausor** [comment]
Merging seems doable, but I don't know if we want our generated classes to have a concrete base class with methods that aren't very useful for our case. The existing ResourceName class could implement our interface, but that doesn't solve our immediate naming issue.

**Comment by garrettjonesgoogle** [comment]
Another option is to rename the existing `ResourceName` class to something else - one option I can think of is `TemplatedResourceName`.

**Comment by michaelbausor** [comment]
Now would be a good time to do this, since we are moving the class from GAX to api-common. Maybe this is the best option - I will use `TemplatedResourceName` unless I think of something else.

**Comment by michaelbausor** [comment]
PTAL

File: src/test/java/com/google/api/pathtemplate/ResourceNameTest.java [hunk 1]
```diff
@@ -47,7 +47,7 @@
   @Test
   public void resourceNameMethods() {
```
Comment (michaelbausor) [inline src/test/java/com/google/api/pathtemplate/ResourceNameTest.java]: Done

**Comment by michaelbausor** [comment]
PTAL

**Pull request merged**